recipes/ignition-math/all/patches/0001-Define-time_regex-locally.patch

```diff
@@ -0,0 +1,42 @@
+From 24a9b3284fcbf05cfec09d9cbd4de4d9977c8318 Mon Sep 17 00:00:00 2001
+From: Juan Oxoby <juan@vicarious.com>
+Date: Fri, 16 Oct 2020 21:28:37 -0700
+Subject: [PATCH] Define time_regex locally
+
+---
+ include/ignition/math/Helpers.hh | 5 +++--
+ 1 file changed, 3 insertions(+), 2 deletions(-)
+
+diff --git a/include/ignition/math/Helpers.hh b/include/ignition/math/Helpers.hh
+index 2dd5cc6..bd4d35f 100644
+--- a/include/ignition/math/Helpers.hh
++++ b/include/ignition/math/Helpers.hh
+@@ -870,7 +870,7 @@ namespace ignition
+     // The following regex takes a time string in the general format of
+     // "dd hh:mm:ss.nnn" where n is milliseconds, if just one number is
+     // provided, it is assumed to be seconds
+-    static const std::regex time_regex(
++    static const char* time_regex_str =
```

Comment:
What's wrong with creating std::regex here?

Reply:
Excellent questions....

From the original author https://github.com/conan-io/conan-center-index/pull/3215#issuecomment-711069238

>After a lot of pain, I was able to solve the segfault error that was happening. The problem was the static initialization of a std::regex variable: https://github.com/ignitionrobotics/ign-math/blob/51eb0640a0c7454c36fef1a3007d3db37fe9165f/include/ignition/math/Helpers.hh#L873, which I solved by creating a raw static char* and using that to initialize a local static (static inside the function) std::regex variable: https://github.com/conan-io/conan-center-index/pull/3215/files#diff-e2f831892c4567a7388d8d62242576015a59f008af8dd34c946996d15f6991e9

Reply:
I don't see what problem it fixes and how, but good to know it fixes something :)

Reply:
💯 I agree... I just didn't want to waste the time we put in so I wanted to keep it but I have no clue what this project is or does.

Reply:
Ok, works for me 👍